packages/cadl-ranch-specs/http/azure/client-generator-core/format/main.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+/**
+ * Describe scenarios for various combinations of unions.
+ */
+@supportedBy("dpg")
+@scenarioService("/azure/client-generator-core/format")
+namespace _Specs_.Azure.ClientGenerator.Core.Format;
+
+model Model {
+  @format("guid")
```

Comment:
What is the goal of this test? `@format` is only to be used as a validation tool which our SDK don't do, I worry that if we even add this test it will lead to believe you should create types for those known formats

Reply:
the goal of this test is kind of creating a common platform for all the formats that every language should properly handle. From each language's perspective, they could have their own implementation for these formats but the wire content must remain the same.

Reply:
Oh and most importantly, MPG tsp will need these format to work properly without breaking changes, therefore I believe it benefits if we have the format we supported in a centralized place.
And also this is the reason we put these test cases under `azure` instead of `type/property`

Reply:
Again those formats are just an indication of the pattern of the string. The type requested here is a string not what the format said. If a different format should be used a known scalar should be created.
please do not do this

Reply:
But during the generation of MPG libraries, we never want undeterministic results, such as if the tsp owner defines its own scalar `azureLocation`, we will not generate that type as `Azure.Core.AzureLocation`.
We want only one specific type defined somewhere in our resourcemanager library to be parsed differently from a random scalar our customer defined.

I heard TCGC now just trims off scalar and returns us its underlying type, if this is true, it is not acceptable for MPG

Reply:
Tcgc should be aware of all the established built in scalar we have and reduce to those not use format.

Reply:
There were one in typespec-azure-core of `eTag`
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/models.tsp#L343-L345

I do not mean this is the expected definition. Just a discussion about whether we should do e.g. `@format(eTag)` or something like it. If we do, should we prefer it is done as `scalar guid is ...` in typespec-azure-resource-manager lib (instead of in service ts files)

Reply:
typespec-azure-resource-manager got `arm-id`
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-resource-manager/lib/models.tsp#L100-L108

Reply:
there is a linter rule telling you not to use `@format` in azure core. Those `@format` on those types are to match to some known openapi/autorest formats but they are not meant to be consumed outside. We found out that we shouldn't even have added `@format` for `eTag` because it isn't a known format to autorest so caused all kind of lint issues.